superset-frontend/src/explore/components/ExploreChartPanel.jsx

```diff
@@ -277,6 +288,17 @@ const ExploreChartPanel = ({
     ],
   );
 
+  const getDatasourceAsSaveableDataset = source => {
+    const dataset = {
```

Comment:
```suggestion
return {
      columns: source.columns,
      name: source?.datasource_name || 'Untitled',
      dbId: source.database.id,
      sql: source?.sql || '',
      schema: source?.schema,
}
```
```

Reply:
THis causes linting errors, so went back to old format.